Automattic/wp-calypso
# Site profiler: Language support

Closes https://github.com/Automattic/wp-calypso/issues/82680

## Proposed Changes

* Supported language provided via route
* Refactored routing configuration

## Testing Instructions



* Go to `/site-profiler`
* Change the language from the Language chooser dropdown in the footer (for example Deutsch)
* Check if language has been change after the page re-render

- Go to `/de/site-profiler` or `/site-profiler/de`
- Check if language has been change

* Go to `/de/site-profiler/asdf.com` or `/site-profiler/de/asdf.com`
* Check if router handled route as before

- Go to `/de/site-profiler?domain=asdf.com` or `/site-profiler/de?domain=asdf.com`
- Check if router handled route as before

* Got to `/ar/site-profiler`
* Check if globe background is flipped

**Before:**
<img width="1728" alt="Screenshot 2023-10-10 at 12 45 49" src="https://github.com/Automattic/wp-calypso/assets/1241413/94d7e2f3-e2a9-4374-89ed-467eacae3427">
**After:**
<img width="1728" alt="Screenshot 2023-10-10 at 12 43 11" src="https://github.com/Automattic/wp-calypso/assets/1241413/500d58ba-dd89-40a9-ba69-230ee8578e4c">

## Pre-merge Checklist
